### PR TITLE
updated tensorflow to 2.7

### LINKED ISSUE #2716: Upgrading to tensorflow 2.6/2.7
A problem with tf 2.5 is that it supports only numpy 1.19 but the latest torch version supports newer version 1.21. Should we upgrade to tensorflow 2.6? (though 2.6 also uses numpy 1.19)

Edit: A broader question I have is that version of Numpy shall deepchem support.

### PR DESCRIPTION
**Pull request by arunppsg**
Fix #2716 

Tensorflow is updated to 2.7.

### CODE REVIEW

**Comment by arunppsg** [comment]
TL; DR:

- CI checks are passing as before. This can be merged.
- We need to consolidate requirements files and their usages (#2782) before further changes to CI workflows/environment files.
_______________________________________________________________
Notes on CI:
- Core CI: failing in windows latest 3.7 (previously, it was also failing for ubuntu latest 3.9 but it passed this time)
- Common CI: failing in windows latest 3.7
- Tensorflow, Torch, Jax: All passing

A couple of pointers:

- The CI for build scripts run only when the *yml files i.e requirements file changes. In this PR, the requirement file changed and it resulted in three additional CI workflows - checking of build on ubuntu, macOS, windows. The CI here is passing only for ubuntu. For macOS,  github actions uses macOS 11 BigSur as the latest version ([ref](https://github.blog/changelog/2021-09-29-github-actions-jobs-running-on-macos-latest-are-now-running-on-macos-big-sur-11/)). I am not sure why it is failing here. Same for windows. The fix for #2781 should get the windows CI running.

- I have a doubt that this PR does not result in any change to CI workflows (the env. in which CI runs) of tensorflow. Two reasons for it:
  - `env.common.yml` is used only in Core CI which does mypy, flake8, yapf, doctests. Hence, I think the change does not touch tensorflow tests.
  - the tensorflow CI described in `tensorflow_setup.yml` uses requirements from   `requirements/tensorflow/env_tensorflow.yml` and `env.test.yml`. Both of these files are not changed as they don't have tensorflow dependencies (and hence they don;t have any effect on tensorflow version in use). I guess that DeepChem's workflow uses tensorflow provided by github actions as part of their python virtual environment.

**Comment by rbharath** [comment]
Capturing the offline discussion from the review, we need to unpin the TF version in `setup.py`:

https://github.com/deepchem/deepchem/blob/master/setup.py#L17

It looks like TF tests are still being run against TF 2.5

**Comment by rbharath** [comment]
@arunppsg I think we have one flaky failure and the usual other failures. I think we can go ahead and merge if this looks good to you

**Comment by arunppsg** [comment]
The one failing in tensorflow CI - ubuntu with python 3.7 is a flaky test. I am tracking these separately and will add a flaky marker later. Hence, we can merge it.
